harness/determined/launch/autodeepspeed.py

```diff
@@ -69,6 +71,34 @@ def create_run_command(master_address: str, hostfile_path: str) -> List[str]:
     return deepspeed_process_cmd
 
 
+@contextlib.contextmanager
+def run_sshd(addrs: List[str], sshd_cmd: List[str]) -> Iterator[None]:
+    """
+    Context manager that runs SSHD and waits for availability, then kills the daemon after exit
+    """
+    # Create the environment file that will be passed by deepspeed to individual ranks.
+    create_deepspeed_env_file()
+    # Set custom PDSH args:
+    # * bypass strict host checking
+    # * -p our custom port
+    # * other args are default ssh args for pdsh
+    os.environ["PDSH_SSH_ARGS"] = (
+        "-o PasswordAuthentication=no -o StrictHostKeyChecking=no "
+        f"-p {constants.DTRAIN_SSH_PORT} -2 -a -x %h"
+    )
+
+    sshd_process = subprocess.Popen(sshd_cmd)
+
+    # Chief machine waits for every worker's sshd to be available.  All machines should be
+    # close to in-step by now because all machines just finished synchronizing rendezvous info.
+    check_sshd(addrs, time.time() + 20, constants.DTRAIN_SSH_PORT)
+    try:
+        yield
+    finally:
+        sshd_process.send_signal(signal.SIGKILL)
```

Comment:
let's use `sshd_process.kill()` instead, which is the more common way to [do this](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.terminate).